modules/programs/broot.nix

```diff
@@ -44,233 +190,41 @@ in {
-          <listitem><para><literal>input</literal></para></listitem>
-          <listitem><para><literal>link</literal></para></listitem>
-          <listitem><para><literal>permissions</literal></para></listitem>
-          <listitem><para><literal>selected_line</literal></para></listitem>
-          <listitem><para><literal>size_bar_full</literal></para></listitem>
-          <listitem><para><literal>size_bar_void</literal></para></listitem>
-          <listitem><para><literal>size_text</literal></para></listitem>
-          <listitem><para><literal>spinner</literal></para></listitem>
-          <listitem><para><literal>status_error</literal></para></listitem>
-          <listitem><para><literal>status_normal</literal></para></listitem>
-          <listitem><para><literal>table_border</literal></para></listitem>
-          <listitem><para><literal>tree</literal></para></listitem>
-          <listitem><para><literal>unlisted</literal></para></listitem>
-        </itemizedlist></para>
-
-        <para>
-        Add <literal>_fg</literal> for a foreground color and
-        <literal>_bg</literal> for a background colors.
-      '';
+      description = "Verbatim config entries";
     };
   };
 
   config = mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
     xdg.configFile."broot/conf.toml".source =
-      tomlFormat.generate "broot-config" brootConf;
+      tomlFormat.generate "broot-config" cfg.settings;
 
     # Dummy file to prevent broot from trying to reinstall itself
     xdg.configFile."broot/launcher/installed-v1".text = "";
 
-    programs.bash.initExtra = mkIf cfg.enableBashIntegration (
-      # Using mkAfter to make it more likely to appear after other
-      # manipulations of the prompt.
-      mkAfter ''
-        # This script was automatically generated by the broot function
-        # More information can be found in https://github.com/Canop/broot
-        # This function starts broot and executes the command
-        # it produces, if any.
-        # It's needed because some shell commands, like `cd`,
-        # have no useful effect if executed in a subshell.
-        function br {
-            f=$(mktemp)
-            (
-                set +e
-                broot --outcmd "$f" "$@"
-                code=$?
-                if [ "$code" != 0 ]; then
-                    rm -f "$f"
-                    exit "$code"
-                fi
-            )
-            code=$?
-            if [ "$code" != 0 ]; then
-                return "$code"
-            fi
-            d=$(cat "$f")
-            rm -f "$f"
-            eval "$d"
-        }
-      '');
+    programs.broot.settings = builtins.fromTOML (builtins.readFile
+      (pkgs.runCommand "default-conf.toml" {
+        nativeBuildInputs = with pkgs; [ hjson remarshal ];
+      } ''
+        hjson -c ${cfg.package.src}/resources/default-conf.hjson \
+        | remarshal --input-format json --output-format toml --output $out
```

Comment:
Is the remashal necessary? Wouldn't `builtins.fromJSON` suffice?

Reply:
Yes! :heavy_check_mark: